extensions/applicationinsights-dependencies-js/src/ajax.ts

```diff
@@ -598,11 +598,15 @@ export class AjaxMonitor extends BaseTelemetryPlugin implements IDependenciesPlu
 
                     if (_enableAjaxPerfTracking) {
                         let iKey = config.instrumentationKey || "unkwn";
-                        if (iKey.length > 5) {
-                            _markPrefix = AJAX_MONITOR_PREFIX + strSubstring(iKey, iKey.length - 5) + ".";
-                        } else {
-                            _markPrefix = AJAX_MONITOR_PREFIX + iKey + ".";
+                        // only change the ikey if it is string
+                        if (isString(iKey)) {
```

Comment:
This is probably going to be an interesting race condition at some point. We might need to change this at some point so it's using the core instance id rather than the iKey.

Reply:
add todo and a task